doc/ansible/dev/developer_guide.md

```diff
@@ -0,0 +1,430 @@
+COPY watches.yaml ${HOME}/watches.yaml
+```
+
+Alternatively, to generate a skeleton project with the above changes, a
+developer can also do:
+```bash
+$ operator-sdk new --type ansible --kind Foo --api-version foo.example.com/v1alpha1 foo-operator --generate-playbook
+```
+
+### Custom Resource file
+
+The Custom Resource file format is Kubernetes resource file. The object has
+mandatory fields:
+
+**apiVersion**:  The version of the Custom Resource that will be created.
+
+**kind**:  The kind of the Custom Resource that will be created
+
+**metadata**:  Kubernetes specific metadata to be created
+
+**spec**:  This is the key-value list of variables which are passed to Ansible.
+This field is optional and will be empty by default.
+
+**annotations**: Kubernetes specific annotations to be appened to the CR. See
+the below section for Ansible Operator specifc annotations.
+
+#### Ansible Operator annotations
+This is the list of CR annotations which will modify the behavior of the operator:
+
+**ansible.operator-sdk/reconcile-period**: Used to specify the reconciliation
+interval for the CR. This value is parsed using the standard Golang package
+[time][time_pkg]. Specifically [ParseDuration][time_parse_duration] is used
+which will use the default of `s` suffix giving the value in seconds.
+
+Example:
+```
+apiVersion: "foo.example.com/v1alpha1"
+kind: "Foo"
+metadata:
+  name: "example"
+annotations:
+  ansible.operator-sdk/reconcile-period: "30s"
+```
+
+### Testing an Ansible operator locally
+
+Once a developer is comfortable working with the above workflow, it will be
+beneficial to test the logic inside of an operator. To accomplish this, we can
+use `operator-sdk up local` from the top-level directory of our project. The
+`up local` command reads from `./watches.yaml` and uses `~/.kube/config` to
+communicate with a kubernetes cluster just as the `k8s` modules do. This
+section assumes the developer has read the [Ansible Operator user
+guide][ansible_operator_user_guide] and has the proper dependencies installed.
+
+Since `up local` reads from `./watches.yaml`, there are a couple options
+available to the developer. If `role` is left alone (by default
+`/opt/ansible/roles/<name>`) the developer must copy the role over to
+`/opt/ansible/roles` from the operator directly. This is cumbersome because
+changes will not be reflected from the current directory. It is recommended
+that the developer instead change the `role` field to point to the current
+directory and simply comment out the existing line:
+```yaml
+- version: v1alpha1
+  group: foo.example.com
+  kind: Foo
+  #  role: /opt/ansible/roles/Foo
+  role: /home/user/foo-operator/Foo
+```
+
+Create a Custom Resource Definiton (CRD) and proper Role-Based Access Control
+(RBAC) defitinions for resource Foo. `operator-sdk` autogenerates these files
+inside of the `deploy` folder:
+```bash
+$ kubectl create -f deploy/crd.yaml
+$ kubectl create -f deploy/rbac.yaml
```

Comment:
This is now I guess `role.yaml`, `role_binding.yaml`, and `service_account.yaml` in the deploy directory.